node/src/Producer.ts

```diff
@@ -603,14 +618,84 @@ export class Producer<ProducerAppData extends AppData = AppData>
 	}
 }
 
+export function producerTypeFromFbs(type: FbsRtpParameters.Type): ProducerType
+{
+	switch (type)
+	{
+		case FbsRtpParameters.Type.SIMPLE:
+			return 'simple';
+		case FbsRtpParameters.Type.SIMULCAST:
+			return 'simulcast';
+		case FbsRtpParameters.Type.SVC:
+			return 'svc';
+		default:
+			throw new TypeError(`invalid ProducerType: ${type}`);
+	}
+}
+
+export function producerTypeToFbs(type: ProducerType): FbsRtpParameters.Type
+{
+	switch (type)
+	{
+		case 'simple':
+			return FbsRtpParameters.Type.SIMPLE;
+
+		case 'simulcast':
+			return FbsRtpParameters.Type.SIMULCAST;
+
+		case 'svc':
+			return FbsRtpParameters.Type.SVC;
+	}
+}
+
+function producerTraceEventTypeToFbs(eventType: ProducerTraceEventType)
+	: FbsProducer.TraceEventType
+{
+	switch (eventType)
+	{
+		case 'keyframe':
+			return FbsProducer.TraceEventType.KEYFRAME;
+		case 'fir':
+			return FbsProducer.TraceEventType.FIR;
+		case 'nack':
+			return FbsProducer.TraceEventType.NACK;
+		case 'pli':
+			return FbsProducer.TraceEventType.PLI;
+		case 'rtp':
+			return FbsProducer.TraceEventType.RTP;
+		default:
+			throw new TypeError(`invalid eventType: ${eventType}`);
+	}
+}
+
+function producerTraceEventTypeFromFbs(eventType: FbsProducer.TraceEventType)
+	: ProducerTraceEventType
+{
+	switch (eventType)
+	{
+		case FbsProducer.TraceEventType.KEYFRAME:
+			return 'keyframe';
+		case FbsProducer.TraceEventType.FIR:
+			return 'fir';
+		case FbsProducer.TraceEventType.NACK:
+			return 'nack';
+		case FbsProducer.TraceEventType.PLI:
+			return 'pli';
+		case FbsProducer.TraceEventType.RTP:
+			return 'rtp';
+		default:
+			throw new TypeError(`invalid TraceEventType: ${eventType}`);
```

Comment:
```suggestion
			throw new TypeError(`invalid FbsProducer.TraceEventType: ${eventType}`);
```

Reply:
I'm removing this line as the switch is already exhausted.